WalletWasabi/Wallets/Wallet.cs

```diff
@@ -258,9 +259,12 @@ public override async Task StartAsync(CancellationToken cancel)
 		bool allowUnconfirmed = false,
 		IEnumerable<OutPoint>? allowedInputs = null,
 		IPayjoinClient? payjoinClient = null,
-		bool tryToSign = true)
+		bool tryToSign = true,
+		bool allowUnavailableCoins = false)
 	{
-		var builder = new TransactionFactory(Network, KeyManager, Coins, BitcoinStore.TransactionStore, password, allowUnconfirmed);
+		var allCoins = allowUnavailableCoins ? Coins : Coins.Available();
```

Comment:
Why are these needed? I think we shouldn't do anything here as the filtering of the coins should happen upper level.

Reply:
I don't know where else should I place this.

The flow goes like this:

First we call this line, here we only use the `_wallet` as whole, so I can't really do any filtering. https://github.com/zkSNACKs/WalletWasabi/blob/f2aeaa4a91afc7b2c8ffb4fdfe21af78b1d43e8b/WalletWasabi.Fluent/ViewModels/Wallets/Send/TransactionPreviewViewModel.cs#L296

Then we go into the `TransactionHelpers.cs`, which we already talked about that it shouldn't contain filtering.

Then we arrive here.

Reply:
I checked the TransactionFactory class because I thought it simply uses the `allowedInputs` and adds the address resued inputs additionally from `allCoins`. But it seems it gathers everything from `allCoins` based on `allowedInputs` (which is not even necessary). My bad.

So [my question](https://github.com/zkSNACKs/WalletWasabi/pull/9706/commits/4d35962241a4a312985ba67bcad4c93abd2d05ce#r1047060573) was false. And you were on the right path in [this commit](https://github.com/zkSNACKs/WalletWasabi/pull/9706/commits/4d35962241a4a312985ba67bcad4c93abd2d05ce).

Reply:
I'm not sure that I was on the right path with that commit.

Isn't it better to build the `TransactionFactory` with the already filtered coins?